test/e2e/node_reuse_test.go

```diff
@@ -193,17 +242,25 @@ func node_reuse() {
 			}
 		}
 	}`)
-	err = targetClusterClient.Patch(ctx, &ctrlplane, client.RawPatch(types.MergePatchType, patch))
-	Expect(err).To(BeNil(), "Failed to set up KCP maxSurge to 1")
+	Eventually(
+		func() error {
+			return targetClusterClient.Patch(ctx, &ctrlplane, client.RawPatch(types.MergePatchType, patch))
+		}, e2eConfig.GetIntervals(specName, "wait-patch")...,
+	).Should(Succeed(), "Failed to set up KCP maxSurge to 1")
+
+	By("Untaint all CP nodes")
+	// The rest of CP nodes may take time to be untaintable
```

Comment:
I am not convinced with this step. In previous step you assure that env has 3 updated CP bmh and machines are in running state.  If CPs is running, it doesn't need untaint loop at all.